File: Competetive programming/Python/Unit 5/A_Multiplication_Game.py
print("===== A MULTIPLICATION GAME =====\n")

n = int(input("Enter Target Number: "))

value = 1
stan_turn = True

while value < n:

    if stan_turn:
        value *= 9
    else:
        value *= 2

    stan_turn = not stan_turn

if not stan_turn:
    print("\nStan wins.")
else:
    print("\nOllie wins.")

#162

File: Competetive programming/C/Unit 5/A_Multiplication_Game.c
#include <stdio.h>

int main() {

    long long n;
    double p = 1;

    printf("===== A MULTIPLICATION GAME =====\n\n");

    printf("Enter target number N: ");
    printf("\nExample: 162\n\n");

    scanf("%lld", &n);

    int stanTurn = 1;

    while(p < n) {

        if(stanTurn)
            p *= 9;   // Stan plays optimally
        else
            p *= 2;   // Ollie plays optimally

        stanTurn = !stanTurn;
    }

    if(!stanTurn)
        printf("\nWinner: Stan\n");
    else
        printf("\nWinner: Ollie\n");

    return 0;
}